openverse_catalog/dags/providers/provider_api_scripts/provider_data_ingester.py

```diff
@@ -100,23 +101,25 @@ def ingest_records(self, **kwargs) -> None:
 
         logger.info(f"Begin ingestion for {self.__class__.__name__}")
 
-        while should_continue:
-            query_params = self.get_next_query_params(query_params, **kwargs)
-
-            batch, should_continue = self.get_batch(query_params)
+        try:
+            while should_continue:
+                query_params = self.get_next_query_params(query_params, **kwargs)
 
-            if batch and len(batch) > 0:
-                record_count += self.process_batch(batch)
-                logger.info(f"{record_count} records ingested so far.")
-            else:
-                logger.info("Batch complete.")
-                should_continue = False
+                batch, should_continue = self.get_batch(query_params)
 
-            if self.limit and record_count >= self.limit:
-                logger.info(f"Ingestion limit of {self.limit} has been reached.")
-                should_continue = False
+                if batch and len(batch) > 0:
+                    record_count += self.process_batch(batch)
+                    logger.info(f"{record_count} records ingested so far.")
+                else:
+                    logger.info("Batch complete.")
+                    should_continue = False
 
-        self.commit_records()
+                if self.limit and record_count >= self.limit:
+                    logger.info(f"Ingestion limit of {self.limit} has been reached.")
+                    should_continue = False
+        finally:
+            total = self.commit_records()
+            logger.info(f"Committed {total} records")
```

Comment:
Interesting that you don't need an `except` clause, and can handle the saved exception after the `finally` clause.

Reply:
This kind of thing I think is usually clearer (at least in the intention communicated by the code) when a context manager is used. I considered suggesting it but it would require factoring code out into another class to act as the manager so it didn't seem worth it. `try/finally` [also works in JavaScript](https://github.com/WordPress/openverse-frontend/blob/b863e3e4b88eb667a87a2c620abc688ab29838eb/src/plugins/api-token.server.ts#L146-L162) and other languages (at least the ones I've used) that have `try/catch/finally` pattern.